From: Tim <NightFlyer@users.noreply.github.com>
Date: Thu, 30 May 2019 18:25:15 -0700
Subject: Change cmark_gfm_extensions_get_tasklist_state to
 cmark_gfm_extensions_tasklist_state_is_checked (#161)

This is in response to issue #160.
This removes cmark_gfm_extensions_get_tasklist_state and adds the
new cmark_gfm_extensions_tasklist_state_is_checked.
---
 extensions/cmark-gfm-core-extensions.h |  3 ++-
 extensions/tasklist.c                  | 15 +++++++++------
 2 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/extensions/cmark-gfm-core-extensions.h b/extensions/cmark-gfm-core-extensions.h
index 8ab049d..126f2c7 100644
--- a/extensions/cmark-gfm-core-extensions.h
+++ b/extensions/cmark-gfm-core-extensions.h
@@ -7,6 +7,7 @@ extern "C" {
 
 #include "cmark-gfm-extension_api.h"
 #include "cmark-gfm-extensions_export.h"
+#include "config.h"
 #include <stdint.h>
 
 CMARK_GFM_EXTENSIONS_EXPORT
@@ -22,7 +23,7 @@ CMARK_GFM_EXTENSIONS_EXPORT
 int cmark_gfm_extensions_get_table_row_is_header(cmark_node *node);
 
 CMARK_GFM_EXTENSIONS_EXPORT
-char *cmark_gfm_extensions_get_tasklist_state(cmark_node *node);
+bool cmark_gfm_extensions_tasklist_state_is_checked(cmark_node *node);
 
 #ifdef __cplusplus
 }
diff --git a/extensions/tasklist.c b/extensions/tasklist.c
index 9caaadb..4af7637 100644
--- a/extensions/tasklist.c
+++ b/extensions/tasklist.c
@@ -9,19 +9,22 @@ typedef enum {
   CMARK_TASKLIST_CHECKED,
 } cmark_tasklist_type;
 
+// Local constants
+static const char *TYPE_STRING = "tasklist";
+
 static const char *get_type_string(cmark_syntax_extension *extension, cmark_node *node) {
-  return "tasklist";
+  return TYPE_STRING;
 }
 
-char *cmark_gfm_extensions_get_tasklist_state(cmark_node *node) {
-  if (!node || ((int)node->as.opaque != CMARK_TASKLIST_CHECKED && (int)node->as.opaque != CMARK_TASKLIST_NOCHECKED))
-    return 0;
+bool cmark_gfm_extensions_tasklist_state_is_checked(cmark_node *node) {
+  if (!node || !node->extension || strcmp(cmark_node_get_type_string(node), TYPE_STRING))
+    return false;
 
   if ((int)node->as.opaque == CMARK_TASKLIST_CHECKED) {
-    return "checked";
+    return true;
   }
   else {
-    return "unchecked";
+    return false;
   }
 }
 
